string MAC
state state
behv beh